lib/OpenQA/WebAPI/Controller/API/V1/Locks.pm

```diff
@@ -18,31 +18,44 @@ use Mojo::Base 'Mojolicious::Controller';
 
 use OpenQA::IPC;
 
-sub mutex_lock {
-    my ($self) = @_;
-    my $name   = $self->stash('name');
-    my $jobid  = $self->stash('job_id');
-    my $ipc    = OpenQA::IPC->ipc;
-    my $res = $ipc->scheduler('mutex_lock', $name, $jobid);
-    return $self->render(text => 'ack', status => 200) if $res;
-    return $self->render(text => 'nack', status => 409);
-}
+sub mutex_action {
+    my ($self)     = @_;
+    my $name       = $self->stash('name');
+    my $jobid      = $self->stash('job_id');
+    my $validation = $self->validation;
 
-sub mutex_unlock {
-    my ($self) = @_;
-    my $name   = $self->stash('name');
-    my $jobid  = $self->stash('job_id');
+    $validation->required('action')->in(qw(lock unlock));
+    $validation->optional('where')->like(qr/^[0-9]+$/);
+    return $self->render(text => 'Bad request', status => 400) if ($validation->has_error);
+
+    my $action = $validation->param('action');
+    my $where  = $validation->param('where');
     my $ipc    = OpenQA::IPC->ipc;
-    my $res = $ipc->scheduler('mutex_unlock', $name, $jobid);
-    return $self->render(text => 'ack', status => 200) if $res;
+
+    my $res;
+    if ($action eq 'lock') {
+        $res = $ipc->scheduler('mutex_lock', $name, $jobid, $where);
```

Comment:
there should be some check that `$where` is set. It's optional in validation, for unlock action, but lock DBus method requires it.